app/src/renderer/vuex/modules/send.js

```diff
@@ -179,12 +194,37 @@ export default ({ node }) => {
 
       // broadcast transaction with signatures included
       const signedTx = state.externals.createSignedTx(tx, signature)
-      const body = state.externals.createBroadcastBody(signedTx)
-      const res = await node.postTx(body).catch(handleSDKError)
+      const body = state.externals.createBroadcastBody(signedTx, `sync`)
+
+      const res = await state.node.postTx(body).catch(handleSDKError)
 
       // check response code
       assertOk(res)
-      commit(`setNonce`, String(parseInt(state.nonce) + 1))
+
+      // sync success
+      if (res.height !== `0`) {
+        commit(`setNonce`, String(parseInt(state.nonce) + 1))
+        return
+      }
+
+      return dispatch(`queryTxInclusion`, res.txhash)
+    },
+    // wait for inclusion of a tx in a block
+    async queryTxInclusion({ state }, txHash) {
+      let success = false
+      let iterations = state.txQueryIterations // 30 * 2s = 60s max waiting time
+      while (!success && iterations-- > 0) {
+        await new Promise(resolve => setTimeout(resolve, state.txQueryTimeout))
+        try {
+          await state.node.tx(txHash)
+          success = true
+        } catch (err) {
+          // tx wasn't included in a block yet
+        }
+      }
+      if (iterations <= 0) {
+        throw new Error(`The transaction was still not included in a block. We can't say for certain it will be included in the future.`)
```

Comment:
shouldn't this be a transaction pending on the Transactions page instead ? that seems like a more clean implementation to me (see designs from Zplin).

Reply:
Step by step by step ;)